### PR TITLE
Chore/mp/confhost

### PR DESCRIPTION
**Pull request by markkopaas**

### CODE REVIEW

File: applications/search/src/api/apis.js [hunk 1]
```diff
@@ -1,12 +1,13 @@
 import axios from 'axios';
+import { getConfig } from '../config';
 
-export const apisUrlBase = '/api/apis';
+export const apisUrlBase = () => `${getConfig().apiApi.host}/api/apis`;
```
Comment (azaxarov) [inline applications/search/src/api/apis.js:4]: This does not need to be a function.

File: applications/search/src/api/apis.js [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
 
-export const apisSearch = params => axios(apisUrlBase, { params }).then(r => r.data);
+export const apisSearch = params => axios(apisUrlBase(), { params }).then(r => r.data);
```
Comment (azaxarov) [inline applications/search/src/api/apis.js:6]: Do we want to catch the result as well?

File: applications/search/src/api/apis.js [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
 
 // NOTE: Response of this function can be mocked using mock files in src/mock
 export const getApi = id =>
-  axios.get(`${apisUrlBase}/${id}`).then(r => r.data).catch(() => null);
+  axios.get(`${apisUrlBase()}/${id}`).then(r => r.data).catch(() => null);
```
Comment (azaxarov) [inline applications/search/src/api/apis.js:10]: Just axios(... instead of axios.get(... ?

File: applications/search/src/api/concepts.js [hunk 4]
```diff
@@ -1,12 +1,13 @@
 import axios from 'axios';
 import get from 'lodash/get';
+import { getConfig } from '../config';
 
-export const conceptsUrlBase = '/api/concepts';
+export const conceptsUrlBase = () => `${getConfig().conceptApi.host}/api/concepts`;
```
Comment (azaxarov) [inline applications/search/src/api/concepts.js:5]: Does not need to be a function.

File: applications/search/src/api/concepts.js [hunk 5]
```diff
@@ -1,12 +1,13 @@
...
 
-export const conceptsSearch = params => axios(conceptsUrlBase, { params }).then(r => r.data);
+export const conceptsSearch = params => axios(conceptsUrlBase(), { params }).then(r => r.data);
 
 export const getConcept = async id =>
-  axios.get(`${conceptsUrlBase}/${id}`).then(r => r.data).catch(() => null);
+  axios.get(`${conceptsUrlBase()}/${id}`).then(r => r.data).catch(() => null);
```
Comment (azaxarov) [inline applications/search/src/api/concepts.js:10]: Just axios(... instead of axios.get(... ?

File: applications/search/src/api/datasets.js [hunk 6]
```diff
@@ -1,18 +1,19 @@
...
-export const datasetsUrlBase = '/datasets';
+export const datasetsUrlBase = () => `${getConfig().datasetApi.host}/datasets`;
 
 export const datasetsSearch = params =>
-  axios(datasetsUrlBase, { params }).then(r => r.data);
+  axios(datasetsUrlBase(), { params }).then(r => r.data);
 
 export const getDataset = id =>
-  axios.get(`${datasetsUrlBase}/${id}`).then(r => r.data).catch(() => null);
+  axios.get(`${datasetsUrlBase()}/${id}`).then(r => r.data).catch(() => null);
```
Comment (azaxarov) [inline applications/search/src/api/datasets.js:12]: Just axios(... instead of axios.get(... ?

File: applications/search/src/api/datasets.js [hunk 7]
```diff
@@ -1,18 +1,19 @@
...
 
 export const getDatasetByURI = uri =>
   axios
-    .get(`${datasetsUrlBase}/byuri`, { params: { uri } }).then(r => r.data).catch(() => null);
+    .get(`${datasetsUrlBase()}/byuri`, { params: { uri } }).then(r => r.data).catch(() => null);
```
Comment (azaxarov) [inline applications/search/src/api/datasets.js:16]: Just axios(... instead of axios.get(... ?

File: applications/search/src/api/datasets.js [hunk 8]
```diff
@@ -1,18 +1,19 @@
 import axios from 'axios';
 import get from 'lodash/get';
 import { normalizeAggregations } from '../lib/normalizeAggregations';
+import { getConfig } from '../config';
 
-export const datasetsUrlBase = '/datasets';
+export const datasetsUrlBase = () => `${getConfig().datasetApi.host}/datasets`;
```
Comment (azaxarov) [inline applications/search/src/api/datasets.js:6]: This does not need to be a function.

File: applications/search/src/api/get-api-stats.js [hunk 9]
```diff
@@ -33,7 +33,7 @@ export function extractStats(data) {
 const statsAggregations = `formats,orgPath,firstHarvested,publisher,openAccess,openLicence,freeUsage`;
 
 export const statsUrl = query =>
-  `${apisUrlBase}${qs.stringify(
+  `${apisUrlBase()}${qs.stringify(
```
Comment (azaxarov) [inline applications/search/src/api/get-api-stats.js:36]: This does not need to be a function.

File: applications/search/src/api/get-concept-stats.js [hunk 10]
```diff
@@ -25,7 +25,7 @@ export function extractStats(data) {
 const statsAggregations = `firstHarvested,publisher`;
 
 export const statsUrl = query =>
-  `${conceptsUrlBase}${qs.stringify(
+  `${conceptsUrlBase()}${qs.stringify(
```
Comment (azaxarov) [inline applications/search/src/api/get-concept-stats.js:28]: This does not need to be a function.

File: applications/search/src/api/get-dataset-stats.js [hunk 11]
```diff
@@ -64,7 +64,7 @@ export function extractStats(data) {
 const statsAggregations = `accessRights,theme,orgPath,provenance,spatial,los,firstHarvested,withDistribution,publicWithDistribution,nonpublicWithDistribution,publicWithoutDistribution,nonpublicWithoutDistribution,withSubject,catalog,opendata,nationalComponent,subject,distributionCountForTypeApi,distributionCountForTypeFeed,distributionCountForTypeFile`;
 
 export const statsUrl = query =>
-  `${datasetsUrlBase}${qs.stringify(
+  `${datasetsUrlBase()}${qs.stringify(
```
Comment (azaxarov) [inline applications/search/src/api/get-dataset-stats.js:67]: This does not need to be a function.

File: applications/search/src/api/get-dataset-stats.js [hunk 12]
```diff
@@ -78,7 +78,7 @@ export const getDatasetStats = orgPath =>
     .catch(e => console.log(JSON.stringify(e))); // eslint-disable-line no-console
 
 export const getDatasetCountsBySubjectUri = query => {
-  const postUrl = `${datasetsUrlBase}/search${qs.stringify(
+  const postUrl = `${datasetsUrlBase()}/search${qs.stringify(
```
Comment (azaxarov) [inline applications/search/src/api/get-dataset-stats.js:81]: This does not need to be a function.

File: applications/search/src/api/informationmodels.js [hunk 13]
```diff
@@ -1,13 +1,14 @@
 import axios from 'axios';
 import get from 'lodash/get';
+import { getConfig } from '../config';
 
-const informationmodelsUrlBase = `/api/informationmodels`;
+const informationmodelsUrlBase = () => `${getConfig().informationmodelApi.host}/api/informationmodels`;
```
Comment (azaxarov) [inline applications/search/src/api/informationmodels.js:5]: This does not need to be a function.

File: applications/search/src/api/informationmodels.js [hunk 14]
```diff
@@ -1,13 +1,14 @@
...
 
 export const informationmodelsSearch = params =>
-  axios.get(informationmodelsUrlBase, { params }).then(r => r.data);
+  axios.get(informationmodelsUrlBase(), { params }).then(r => r.data);
```
Comment (azaxarov) [inline applications/search/src/api/informationmodels.js:8]: - axios(...
- url does not need to be a function.

File: applications/search/src/api/publishers.js [hunk 15]
```diff
@@ -0,0 +1,9 @@
+import axios from 'axios';
+import { getConfig } from '../config';
+import get from 'lodash/get';
+
+export const publishersUrlBase = () => `${getConfig().publisherApi.host}/publisher`;
```
Comment (azaxarov) [inline applications/search/src/api/publishers.js]: This does not need to be a function.

File: applications/search/src/api/referenceData.js [hunk 16]
```diff
@@ -0,0 +1,6 @@
+import axios from 'axios';
+import { getConfig } from '../config';
+
+export const referenceDataUrlBase = () => `${getConfig().referenceDataApi.host}/reference-data/`;
```
Comment (azaxarov) [inline applications/search/src/api/referenceData.js]: This does not need to be a function.

Trailing slash?

File: applications/search/src/api/referenceData.js [hunk 17]
```diff
@@ -0,0 +1,6 @@
...
+
+export const getReferenceData = path => axios(referenceDataUrlBase() + path).then(r => r.data);
```
Comment (azaxarov) [inline applications/search/src/api/referenceData.js]: maybe use template string?

**Review by azaxarov — CHANGES_REQUESTED** [review]

File: applications/search/src/api/concepts.js (hunk 5 above)
Comment (markkopaas) [reply applications/search/src/api/concepts.js:10]: det litt mer eksplisitt  med .get :D skal ikke fikse

File: applications/search/src/api/referenceData.js (hunk 16 above)
Comment (markkopaas) [reply applications/search/src/api/referenceData.js]: er det bedre med path som begynner med slash? eller kanskje, query som har base + / + path?

File: applications/search/src/api/referenceData.js (hunk 17 above)
Comment (markkopaas) [reply applications/search/src/api/referenceData.js]: ok, kanskje template skal ha / da

File: applications/search/src/api/apis.js (hunk 2 above)
Comment (markkopaas) [reply applications/search/src/api/apis.js:6]: jeg tenker at klienten skulle gjøre catch, men jeg har ikke sjekket alle sted hvor det kalles... skal det nå

**Review by azaxarov — APPROVED** [review]

**Pull request merged**